### PR TITLE
open system scenario

### PR DESCRIPTION
**Pull request by metar520**
Add SystemService and ExternalServiceHandler with interface

- Implement SystemService class to handle system opening functionality.
- Create ExternalServiceHandler to manage connections to external services.
- Define an interface for external services to standardize interactions.

### CODE REVIEW

File: src/main/java/Domain/ExternalServices/ExternalService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+package Domain.ExternalServices;
+
+// An interface for all external services
+public interface ExternalService {
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ExternalServices/ExternalService.java:5]: great!

File: src/main/java/Domain/ExternalServices/ExternalServiceHandler.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class represents an external service handler.
+ * It manages a list of external services.
+ */
+public class ExternalServiceHandler {
+
+    private List<ExternalService> externalServices;
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ExternalServices/ExternalServiceHandler.java:12]: _externalServices

File: src/main/java/Domain/ExternalServices/SupplyService/AdapterSupply.java [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+
+public class AdapterSupply implements ExternalService{
+
+    public AdapterSupply() {
+    
+    }
+
+    @Override
+    public boolean ConnectToService() {
+        // Connect to the supply service
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ExternalServices/SupplyService/AdapterSupply.java:13]: the addapter wants to call the implement and try to connect - tal is doing that?
if so- leave him //TODO pls

File: src/main/java/Domain/User.java [hunk 4]
```diff
@@ -3,12 +3,14 @@ public class User {
     private String encoded_password;
     private String username;
     private boolean loggedIn;
+    private boolean isAdmin;
```
Comment (inbarbenchaim) [inline src/main/java/Domain/User.java:6]: _isAdmin

Comment (inbarbenchaim) [reply src/main/java/Domain/User.java:6]: if you can change all the rest also pls, if not its ok

File: src/main/java/Domain/ExternalServices/PaymentService/AdapterPayment.java [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+
+public class AdapterPayment implements ExternalService {
+    
+    public AdapterPayment() {
+    
+    }
+
+    @Override
+    public boolean ConnectToService() {
+        // Connect to the payment service
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ExternalServices/PaymentService/AdapterPayment.java:13]: add log print pls

File: src/main/java/Domain/User.java (hunk 4 above)
Comment (metar520) [reply src/main/java/Domain/User.java:6]: It's not a problem, but why? We agreed that the convention is to use lowercase letters for all fields.

File: src/main/java/ServiceLayer/UserService.java [hunk 6]
```diff
@@ -61,4 +61,25 @@ public Response register(String userName, String password, String email) {
...
         return response;
     }
+
+    // function that check if the user is an admin using try catch and logging without checking for token
+    public Response isAdmin(String userName, String password){
+        Response response = new Response();
+        try {
+            if (userController.isAdmin(userName)){
+                logger.info("User is an admin: " + userName);
+                response.setReturnValue("User is an admin");
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/UserService.java:71]: the return value is an object. i think maybe to return the user?

File: src/main/java/ServiceLayer/SystemService.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
+package ServiceLayer;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import Domain.ExternalServices.ExternalServiceHandler;
+
+// Class that represents the system service and enables users (probably admins) to control the system.
+
+public class SystemService {
+     private UserService userService;
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/SystemService.java:9]: _userService and so on

File: src/main/java/ServiceLayer/SystemService.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
+package ServiceLayer;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import Domain.ExternalServices.ExternalServiceHandler;
+
+// Class that represents the system service and enables users (probably admins) to control the system.
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/SystemService.java:6]: good

File: src/main/java/ServiceLayer/SystemService.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+     * 
+     * @param userId   the user ID
+     * @param password the user password
+     * @return a response indicating the success or failure of opening the system
+     */
+    public Response openSystem(String userId, String password) {
+        Response response = new Response();
+        try {
+            // Perform user login
+            Response loginResponse = userService.logIn(userId, password);
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/SystemService.java:30]: i think its happend after he does login, maybe just check he is logged in in boolean in user class

**Review by inbarbenchaim — APPROVED** [review]

**Pull request merged**